src/main/java/io/a2a/spec/JSONRPCRequest.java

```diff
@@ -16,9 +16,9 @@
  */
 @JsonInclude(JsonInclude.Include.NON_ABSENT)
 @JsonIgnoreProperties(ignoreUnknown = true)
-public sealed class JSONRPCRequest implements JSONRPCMessage permits SendTaskRequest, SendTaskStreamingRequest,
+public sealed class JSONRPCRequest implements JSONRPCMessage permits SendTaskStreamingRequest,
         GetTaskRequest, CancelTaskRequest, SetTaskPushNotificationRequest, GetTaskPushNotificationRequest,
-        TaskResubscriptionRequest {
```

Comment:
does it make sense to be non-abstract?
Can we have a JSONRPCRequest which is none of the permitted subclasses?

Reply:
I haven't come across usages yet where a JSONRPCRequest isn't one of the permitted subclasses. I'll make it abstract.